### PR TITLE
Adding tempeh (tem · pay)

### PR DESCRIPTION
**Pull request by kreindo**
## About The Pull Request

I'm adding a new food to the game called tempeh, it's originated from Indonesia. Made by fermenting soybean using fungus (Rhizopus oligosporus or Rhizopus oryzae, probably the in-game way to make tempeh is by using any type of fungus combined with soybean in a bowl), I figured it's a good addition, Currently it's just a copied tofu and change a few variable, further help/assistance may be required to realize this in-game :)

here's a tempeh:
image

## Why It's Good For The Game
It's good for the game mainly content addition and cultural diversity, perhaps for people to learn about other culture / a window to other culture. also tempeh's awesome and healthy. can't have tofu without tempeh

## Changelog

:cl:
add: Added Tempeh
/:cl:

### CODE REVIEW

**Comment by Shadowflame909** [comment]
what does it taste like. Sweet? Salty? Bland?

**Comment by kreindo** [comment]
> what does it taste like. Sweet? Salty? Bland?



> what does it taste like. Sweet? Salty? Bland?

tempeh has a nutty, earthy taste that can vary from mild to strong, depending on the maturity. sometimes people fry them, sometiume people marinates them. some call it the vegan meat

**Comment by kreindo** [comment]
> You didn’t add a way to make this food in game

I suppose i didn't, I kinda just place it there for now, I'm looking at how to add stuff in-game so i can further revise this

**Comment by Wallemations** [comment]
If you're still working on the main aspect of your PR, usually you need to set it as a draft.

As for how to make foods, most are simple recipes. I'd recommend looking up existing food recipes in the code and then going off from there.

Also, your description feels rather bland, mainly in that it's just 4 words. Something like "A cake made from soybeans fermented in fungus." would work well, although still not be very exciting.

**Comment by Wallemations** [comment]
> > what does it taste like. Sweet? Salty? Bland?
> 
> tempeh has a nutty, earthy taste that can vary from mild to strong, depending on the maturity. sometimes people fry them, sometiume people marinates them. some call it the vegan meat

The reason they asked this is the "tastes" list is supposed to tell the player this.
As an example, corned beef tells the player when they eat it that it tastes like meat and cabbage. Simply repeating the name of the food (in most cases) won't be very descriptive.

**Comment by Wallemations** [comment]
I'm not a maintainer, but here's what I'll recommend you do, since this is a pretty simple one PR:

- Set this PR to be a draft. You can google how to do that ("how to set GitHub pull request to draft")
- Add a way to make this food (this could either be a reagent recipe like tofu or a simple crafting menu recipe like most other foods)
- Make the food desc & tastes more unique
- Make a sprite for the food

Once you've done all that, then undraft it. As the PR stands in this moment, it is completely unfinished.

**Comment by kreindo** [comment]
> I'm not a maintainer, but here's what I'll recommend you do, since this is a pretty simple one PR:
> 
> * Set this PR to be a draft. You can google how to do that ("how to set GitHub pull request to draft")
> * Add a way to make this food (this could either be a reagent recipe like tofu or a simple crafting menu recipe like most other foods)
> * Make the food desc & tastes more unique
> * Make a sprite for the food
> 
> Once you've done all that, then undraft it. As the PR stands in this moment, it is completely unfinished.

Alrighty, thanks for the help and suggestion, i'll draft this PR.

File: code/modules/food_and_drinks/recipes/processor_recipes.dm
Comment (EOBGames) [inline code/modules/food_and_drinks/recipes/processor_recipes.dm:93]: This recipe won't work, soybeans already process into soy dope via the food processor (see /datum/food_processor_process/soybeans).

Comment (kreindo) [reply code/modules/food_and_drinks/recipes/processor_recipes.dm:93]: > This recipe won't work, soybeans already process into soy dope via the food processor (see /datum/food_processor_process/soybeans).

yes, I'm planning on changing the recipe, so instead of just soybean, it needs a mix of soybean and mushroom seed (maybe by mixing them in a bowl), then food process that mix instead.

Comment (kreindo) [reply code/modules/food_and_drinks/recipes/processor_recipes.dm:93]: I added the tempeh blend item for food processor instead of soybean

**Comment by kreindo** [comment]
how can i make (mushroom seed + soybean = tempeh) recipe?

**Comment by Wallemations** [comment]
> how can i make (mushroom seed + soybean = tempeh blend) recipe?

You'd have to make it a normal crafting recipe instead of a reagent one. Look at other foods for examples (I.e. burgers or sandwiches)

**Review by Twaticus — APPROVED** [review]

**Comment by kreindo** [comment]
I think the PR is done, although currently I'm using plump seed for the tempeh starter recipe, originally I want it to able to use any mushroom seed, but don't know for sure how to do that, any feedback?

File: code/modules/food_and_drinks/recipes/tablecraft/recipes_meat.dm
Comment (JohnFulpWillard) [inline code/modules/food_and_drinks/recipes/tablecraft/recipes_meat.dm:74]: are you sure this is a meat?

Comment (kreindo) [reply code/modules/food_and_drinks/recipes/tablecraft/recipes_meat.dm:74]: it's like tofu.

**Comment by optimumtact** [comment]
> I think the PR is done, although currently I'm using plump seed for the tempeh starter recipe, originally I want it to able to use any mushroom seed, but don't know for sure how to do that, any feedback?

It's merge conflicting, we can't merge it in this state

**Comment by kreindo** [comment]
> > I think the PR is done, although currently I'm using plump seed for the tempeh starter recipe, originally I want it to able to use any mushroom seed, but don't know for sure how to do that, any feedback?
> 
> It's merge conflicting, we can't merge it in this state
> 
> https://tgstation13.org/wiki/Resolving_icon_conflicts

Ok I'll fix it in a bit

**Comment by SuperNovaa41** [comment]
Fix conflicts and i'll merge

**Pull request merged**
